extensions/wikia/HTTPSSupport/tests/HTTPSSupportHooksTest.php

```diff
@@ -86,4 +90,98 @@ public function onBeforeResourceLoaderCSSMinifierDataProvider() {
+		$this->unsetGlobals();
+		$this->assertEquals( $expectedResult, $url );
+
+	}
+
+	public function onLinkerMakeExternalLinkDataProvider() {
+		return [
+			[
+				'$url' => 'http://www.example.com',
+				'$environment' => WIKIA_ENV_PROD,
+				'$expectedResult' => 'http://www.example.com',
+			],
+			[
+				'$url' => 'https://www.example.com' ,
+				'$environment' => WIKIA_ENV_PROD,
+				'$expectedResult' => 'https://www.example.com',
+			],
+			[
+				'$url' => 'http://ja.starwars.wikia.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_PROD,
+				'$expectedResult' => 'http://ja.starwars.wikia.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://starwars.wikia.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_PROD,
+				'$expectedResult' => 'https://starwars.wikia.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://starwars.fandom.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_PROD,
+				'$expectedResult' => 'https://starwars.fandom.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://starwars.verify.fandom.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_VERIFY,
+				'$expectedResult' => 'https://starwars.verify.fandom.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://starwars.preview.fandom.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_PREVIEW,
+				'$expectedResult' => 'https://starwars.preview.fandom.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://starwars.preview.wikia.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_PREVIEW,
+				'$expectedResult' => 'https://starwars.preview.wikia.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://ja.starwars.preview.fandom.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_PREVIEW,
+				'$expectedResult' => 'https://ja.starwars.preview.fandom.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://ja.starwars.preview.wikia.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_PREVIEW,
+				'$expectedResult' => 'http://ja.starwars.preview.wikia.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://starwars.sandbox-s2.fandom.com/wiki/Yoda?key=value' ,
+				'$environment' => WIKIA_ENV_SANDBOX,
+				'$expectedResult' => 'https://starwars.sandbox-s2.fandom.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://starwars.mockdevname.fandom-dev.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_DEV,
+				'$expectedResult' => 'https://starwars.mockdevname.fandom-dev.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://starwars.mockdevname.wikia-dev.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_DEV,
+				'$expectedResult' => 'https://starwars.mockdevname.wikia-dev.com/wiki/Yoda?key=value',
+			],
+			[
+				'$url' => 'http://ja.starwars.mockdevname.wikia-dev.com/wiki/Yoda?key=value',
+				'$environment' => WIKIA_ENV_DEV,
+				'$expectedResult' => 'https://ja.starwars.mockdevname.wikia-dev.com/wiki/Yoda?key=value',
```

Comment:
I believe the result should be `http://ja.starwars.mockdevname.wikia-dev.com/wiki/Yoda?key=value`